pkg/goDB/godb_test.go

```diff
@@ -0,0 +1,151 @@
+)
+
+func TestWorkloadCreation(t *testing.T) {
+
+	// Initialize temporary test directory
+	testPath, err := os.MkdirTemp("/tmp/test_db", "goDB")
+	require.Nil(t, err)
+	defer os.RemoveAll(testPath)
+	require.Nil(t, os.Mkdir(filepath.Join(testPath, "eth0"), 0700))
+
+	t.Run("empty", func(t *testing.T) {
+		workMgr, err := NewDBWorkManager(testPath, "eth0", 1)
+		require.Nil(t, err)
+
+		nonempty, err := workMgr.CreateWorkerJobs(
+			time.Date(1970, time.January, 1, 0, 0, 0, 0, time.UTC).Unix(),
+			time.Date(1999, time.December, 31, 0, 0, 0, 0, time.UTC).Unix(),
+			&Query{},
+		)
+		require.Nil(t, err)
+		require.False(t, nonempty)
+	})
+
+	// Create some dummy data / directories
+	for day := 1; day < 27; day++ {
+		f := gpfile.NewDir(filepath.Join(testPath, "eth0"), time.Date(2000, time.January, day, 0, 0, 0, 0, time.UTC).Unix(), gpfile.ModeWrite)
+		require.Nil(t, f.Open())
+		require.Nil(t, f.WriteBlocks(time.Date(2000, time.January, day, 0, 0, 0, 0, time.UTC).Unix(), gpfile.TrafficMetadata{}, types.Counters{}, [types.ColIdxCount][]byte{}))
+		require.Nil(t, f.Close())
+
+		f = gpfile.NewDir(filepath.Join(testPath, "eth0"), time.Date(2001, time.January, day, 0, 0, 0, 0, time.UTC).Unix(), gpfile.ModeWrite)
+		require.Nil(t, f.Open())
+		require.Nil(t, f.WriteBlocks(time.Date(2001, time.January, day, 0, 0, 0, 0, time.UTC).Unix(), gpfile.TrafficMetadata{}, types.Counters{}, [types.ColIdxCount][]byte{}))
+		require.Nil(t, f.Close())
+
+		f = gpfile.NewDir(filepath.Join(testPath, "eth0"), time.Date(2001, time.February, day, 0, 0, 0, 0, time.UTC).Unix(), gpfile.ModeWrite)
+		require.Nil(t, f.Open())
+		require.Nil(t, f.WriteBlocks(time.Date(2001, time.February, day, 0, 0, 0, 0, time.UTC).Unix(), gpfile.TrafficMetadata{}, types.Counters{}, [types.ColIdxCount][]byte{}))
+		require.Nil(t, f.Close())
+
+		f = gpfile.NewDir(filepath.Join(testPath, "eth0"), time.Date(2001, time.March, day, 0, 0, 0, 0, time.UTC).Unix(), gpfile.ModeWrite)
+		require.Nil(t, f.Open())
+		require.Nil(t, f.WriteBlocks(time.Date(2001, time.March, day, 0, 0, 0, 0, time.UTC).Unix(), gpfile.TrafficMetadata{}, types.Counters{}, [types.ColIdxCount][]byte{}))
+		require.Nil(t, f.Close())
+
+		f = gpfile.NewDir(filepath.Join(testPath, "eth0"), time.Date(2004, time.January, day, 0, 0, 0, 0, time.UTC).Unix(), gpfile.ModeWrite)
+		require.Nil(t, f.Open())
+		require.Nil(t, f.WriteBlocks(time.Date(2004, time.January, day, 0, 0, 0, 0, time.UTC).Unix(), gpfile.TrafficMetadata{}, types.Counters{}, [types.ColIdxCount][]byte{}))
+		require.Nil(t, f.Close())
+	}
+
+	t.Run("before range", func(t *testing.T) {
+		workMgr, err := NewDBWorkManager(testPath, "eth0", 1)
+		require.Nil(t, err)
+
+		nonempty, err := workMgr.CreateWorkerJobs(
+			time.Date(1970, time.January, 1, 0, 0, 0, 0, time.UTC).Unix(),
+			time.Date(1999, time.December, 31, 0, 0, 0, 0, time.UTC).Unix(),
+			&Query{},
+		)
+		require.Nil(t, err)
+		require.False(t, nonempty)
+	})
+
+	t.Run("after range", func(t *testing.T) {
```

Comment:
Good that this logic is tested intensely.

You could get rid of quite a bit of copy-pasta if you stuff this into a table-driven test. Even if there is a small difference in the `require`-clauses, it might be wort the effort.